bin/circuit_transpiler/src/main.cpp

```diff
@@ -0,0 +1,495 @@
+    using dist_type = std::uniform_int_distribution<int>;
+    static std::random_device random_engine;
+
+    std::vector<std::size_t> step_list;
+    std::size_t steps_sum = 0;
+    while (steps_sum != r) {
+        if (r - steps_sum <= max_step) {
+            while (r - steps_sum != 1) {
+                step_list.emplace_back(r - steps_sum - 1);
+                steps_sum += step_list.back();
+            }
+            step_list.emplace_back(1);
+            steps_sum += step_list.back();
+        } else {
+            step_list.emplace_back(dist_type(1, max_step)(random_engine));
+            steps_sum += step_list.back();
+        }
+    }
+    return step_list;
+}
+
+template<typename FRIScheme, typename FieldType>
+typename FRIScheme::params_type create_fri_params(std::size_t degree_log, const int max_step = 1) {
+    typename FRIScheme::params_type params;
+    nil::crypto3::math::polynomial<typename FieldType::value_type> q = {0, 0, 1};
+
+    constexpr std::size_t expand_factor = 0;
+    std::size_t r = degree_log - 1;
+
+    std::vector<std::shared_ptr<nil::crypto3::math::evaluation_domain<FieldType>>> domain_set =
+        nil::crypto3::math::calculate_domain_set<FieldType>(degree_log + expand_factor, r);
+
+    params.r = r;
+    params.D = domain_set;
+    params.max_degree = (1 << degree_log) - 1;
+    params.step_list = generate_random_step_list(r, max_step);
+
+    return params;
+}
+
+template<typename TIter>
+void print_hex_byteblob(std::ostream &os, TIter iter_begin, TIter iter_end, bool endl) {
+    os << "0x" << std::hex;
+    for (TIter it = iter_begin; it != iter_end; it++) {
+        os << std::setfill('0') << std::setw(2) << std::right << int(*it);
+    }
+    os << std::dec;
+    if (endl) {
+        os << std::endl;
+    }
+}
+
+template<typename Endianness, typename Proof>
+void proof_print(Proof &proof, std::string output_file) {
+    using namespace nil::crypto3::marshalling;
+
+    using TTypeBase = nil::marshalling::field_type<Endianness>;
+    using proof_marshalling_type = nil::crypto3::zk::snark::placeholder_proof<TTypeBase, Proof>;
+    auto filled_placeholder_proof = types::fill_placeholder_proof<Endianness, Proof>(proof);
+
+    std::vector<std::uint8_t> cv;
+    cv.resize(filled_placeholder_proof.length(), 0x00);
+    auto write_iter = cv.begin();
+    nil::marshalling::status_type status = filled_placeholder_proof.write(write_iter, cv.size());
+    std::ofstream out;
+    out.open(output_file);
+    print_hex_byteblob(out, cv.cbegin(), cv.cend(), false);
+}
+
+int main(int argc, char *argv[]) {
+    if (argc < 2) {
+        std::cout << "There are two commands:" << std::endl
+                  << "gen_test_proof prepares gate argument, placeholder params and sample proof for testing"
+                  << std::endl
+                  << "gen_gate_argument prepares gate argument and some placeholder params" << std::endl;
```

Comment:
Is this a help string for something like `transpiler --help`? Let's work on it a little more.

First, what are "some" placeholder params? Does the first command prepare a full set of params?

Second, what is the usage scenario? When will user see this message?